src/hooks/useConnectCurrentEvent.ts

```diff
@@ -22,7 +20,7 @@ const useConnectCurrentEvent = () => {
       collection: "purchases",
       where: [
         ["eventId", "==", eventId],
-        ["userId", "==", user.uid],
+        ["userId", "==", user?.uid],
```

Comment:
Firebase will throw an error if given `undefined` here

Reply:
that is what we want it to do. Can't put a if(!user)return due to hooks. Alternative is to provide an argument to the hook as (userId:string) and only call it in a component where user is defined, but this seems unnecessary.